File: 1207.unique-number-of-occurrences/solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> counts = new HashMap<>();
        Set<Integer> occurrences = new HashSet<>();
        
        for (int val : arr) {
            counts.put(val, counts.getOrDefault(val, 0) + 1);
        }
        
        for (Integer cnt : counts.values()) {
            if (occurrences.contains(cnt)) {
                return false;
            }
            occurrences.add(cnt);
        }
        return true;
    }
}

File: 1207.unique-number-of-occurrences/solution.py
from collections import Counter
from typing import List

class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        counts = Counter(arr)
        occurrences = set()
        for cnt in counts.values():
            if (cnt in occurrences):
                return False
            occurrences.add(cnt)
        return True